SPHINXsys/src/for_3D_build/particles/solid_particles_supplementary.cpp

```diff
@@ -11,7 +11,7 @@ namespace SPH
 	{
 
 		Mat3d F = F_[particle_i];
-		Mat3d epsilon = 0.5 * (~F * F - Matd(1.0)); // calculation of the Green-Lagrange strain tensor
+		Mat3d epsilon = 0.5 * (F.transpose() * F - Matd::Identity()); // calculation of the Green-Lagrange strain tensor
```

Comment:
@ChiZhangatTUM  there is a mix of Matd and Mat3d in this file

Reply:
They are identical. 
There are mix usage of vector or matrix in dimensions, we can use a refactoring branch to clean them, and define a principle to follow.